### PR TITLE
react: implement exercise

### LINKED ISSUE #743: react: implement exercise
There are exercises that are not yet implemented in Python track (the full list of them - https://github.com/exercism/python/issues/417#issuecomment-334841778).
You can help us change this situation by implementing this one. Ref #417

The guide ["Implement an exercise from specification"](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md) will help you succeed.

If you still have questions, I encourage you to ask them before starting working on the issue.

I also suggest claiming the issue by creating an empty PR as stated [here](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#avoiding-duplicate-work)

Please note that there are some special ['closing issue' keywords](https://help.github.com/articles/closing-issues-using-keywords/) that you can put in description of your PR and Github will take care of closing the corresponding issue automatically after merge.
This simple thing helps us a lot, and makes our life a bit easier

### PR DESCRIPTION
**Pull request by cmccandless**
Resolves #743 

TODO:
- [x] add README
- [x] update `config.json`
- [x] create solution template
- [x] write test cases
- [x] write example solution

### CODE REVIEW

**Comment by cmccandless** [comment]
Note: no canonical data exists for this exercise. The test cases were ported from the [C# track](https://github.com/exercism/csharp/blob/5b22854e6858f1a5a95f6d27207523b227f0dd39/exercises/react/ReactTest.cs).

File: exercises/react/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Implement a basic reactive system with cells with settable values ("input"
+cells) and cells with values computed in terms of other cells ("compute"
+cells). Implement updates so that when an input value is changed, values
+propagate to reach a new stable system state.
+
+In addition, compute cells should allow for registering change notification
+callbacks.  Call a cell’s callbacks when the cell’s value in a new stable
+state has changed from the previous stable state.
+
+### Submitting Exercises
```
Comment (N-Parsons) [inline exercises/react/README.md]: This should be at heading level 2.

File: exercises/react/react.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Cell(object):
+    def set_value(value):
+        pass
+
+
+class Reactor(object):
+    def create_input_cell(value):
+        pass
+
+    def create_compute_cell(dependencies, updater):
```
Comment (N-Parsons) [inline exercises/react/react.py]: Presumably these three methods should have `self` as an argument?

File: exercises/react/example.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                 updater=None):
+        self.reactor = reactor
+        self.index = len(reactor.cells)
+        reactor.cells.append(self)
+        self.value = value
+        self.dirty = False
+        self.dependencies = dependencies
+        self.dependents = set()
+        self.updater = updater
+        self.watchers = set()
```
Comment (N-Parsons) [inline exercises/react/example.py]: If you're expecting this to exist (which line 80 of the tests implies), you need to either document it or supply it in the template code. I think you need to revisit this exercise and work out what methods/attributes the tests require to exist and add them to the template or document their expected behaviour. Currently, someone would have to delve into the test-suite to work out what methods should exist.

**Comment by cmccandless** [comment]
@N-Parsons I've address your comments, please review new changes after Travis-CI build has completed. Thanks.

File: exercises/react/example.py (hunk 3 above)
Comment (cmccandless) [reply exercises/react/example.py]: My original intent was to hide that functionality behind methods `add_watcher` and `remove_watcher`; I'm not sure why I did away with this, but I've restored those methods. `Reactor.watchers` is no longer needed by `react_test.py`

File: exercises/react/react.py (hunk 2 above)
Comment (cmccandless) [reply exercises/react/react.py]: Yep. Thanks.

File: exercises/react/README.md (hunk 1 above)
Comment (cmccandless) [reply exercises/react/README.md]: Done.

**Pull request merged**